File: 1716.计算力扣银行的钱.py
#
# @lc app=leetcode.cn id=1716 lang=python3
#
# [1716] 计算力扣银行的钱
#

# @lc code=start
class Solution:
    def totalMoney(self, n: int) -> int:
        res = 0
        currDay = 0 # 虚拟一个第零天，后续相加
        for i in range(n):
            currDay += 1
            if i % 7 == 0 and i != 0:
                currDay -= 6    # -7+1 ===> -6
            res += currDay
        return res
# @lc code=end

File: 1716.计算力扣银行的钱.go
/*
 * @lc app=leetcode.cn id=1716 lang=golang
 *
 * [1716] 计算力扣银行的钱
 */

// @lc code=start
func totalMoney(n int) (res int) {
	currDay := 0 // 模拟一个第零天，后续相加即可
	for i := 0; i < n; i++ {
		currDay++
		if i%7 == 0 && i != 0 {
			currDay -= 6 // currDay = currDay - 7 + 1
		}
		res += currDay
	}
	return res
}

// @lc code=end

